test/unit-tests/ima_test.js

```diff
@@ -417,6 +417,55 @@ describe('ad_manager_ima', function()
     expect(notified).to.be(true);
   });
 
+  it('AMC Integration, Ad Rules: Non-linear ad should trigger forceAdToPlay on AMC', function()
+  {
+    var triggered = 0;
+    google.ima.linearAds = false;
+    amc.forceAdToPlay = function(name, metadata, type)
+    {
+      triggered += 1;
+    };
+    initAndPlay(true, vci);
+    var am = google.ima.adManagerInstance;
+    am.publishEvent(google.ima.AdEvent.Type.AD_BREAK_READY);
+    am.publishEvent(google.ima.AdEvent.Type.LOADED);
+    am.publishEvent(google.ima.AdEvent.Type.STARTED);
+    expect(triggered).to.be(1);
+  });
+
+  it('AMC Integration, Ad Rules: Existing non-linear ad should be cancelled when next ad break is ready', function()
+  {
+    google.ima.linearAds = false;
+    amc.playAd = function (ad) {
+      ima.playAd(ad);
+    };
+    amc.forceAdToPlay = function(name, metadata, type) {
+      var adData = {
+        "adManager": name,
+        "adType": type,
+        "ad": metadata,
+        "streams": {},
+        "position": -1 // play immediately
+      };
+      var adPod = new amc.Ad(adData);
+      adPod.id = "adId";
+      amc.timeline.push(adPod);
+      amc.playAd(amc.timeline.shift());
+    };
+    initAndPlay(true, vci);
+    var am = google.ima.adManagerInstance;
+    am.publishEvent(google.ima.AdEvent.Type.AD_BREAK_READY);
+    am.publishEvent(google.ima.AdEvent.Type.LOADED);
+    am.publishEvent(google.ima.AdEvent.Type.STARTED);
+    // These should exist when overlay is being displayed
+    expect(ima.currentAMCAdPod).to.be.ok();
+    expect(ima.currentIMAAd).to.be.ok();
+    am.publishEvent(google.ima.AdEvent.Type.AD_BREAK_READY);
+    // These should be removed when next ad break is ready
```

Comment:
Yes, this is mostly because there is no `COMPLETE` event for non-linear ads, so there is no way to know when the ad is removed. It's also a prerequisite for this fix that the current ad gets removed before the next ad begins because we need to have `_uiContainer` in its default state.